scripts/interopTestClient.ts

```diff
@@ -0,0 +1,244 @@
+                throw new Error("Measurement is not 0 or 1");
+            }
+
+            vdafObject = {
+                type: vdaf.type,
+            };
+            measurement = body.measurement;
+            break;
+
+        case "Prio3Aes128Sum":
+            if (!("bits" in vdaf)) {
+                throw new Error("VDAF definition is missing number of bits");
+            }
+            if (typeof vdaf.bits !== "number") {
+                throw new Error("VDAF definition's `bits` parameter is not a number");
+            }
+            if (typeof body.measurement !== "string") {
+                throw new Error("Measurement is not a string");
+            }
+
+            vdafObject = {
+                type: vdaf.type,
+                bits: vdaf.bits,
+            };
+            measurement = body.measurement;
+            break;
+
+        case "Prio3Aes128Histogram":
+            if (!("buckets" in vdaf)) {
+                throw new Error("VDAF definition is missing buckets");
+            }
+            if (!Array.isArray(vdaf.buckets)) {
+                throw new Error("VDAF definition's `buckets` parameter is not an array");
+            }
+            for (const bucketBoundary of vdaf.buckets) {
+                if (typeof bucketBoundary !== "string") {
+                    throw new Error("VDAF defeinition's `buckets` parameter is not an array of strings");
+                }
+            }
+            if (typeof body.measurement !== "string") {
+                throw new Error("Measurement is not a string");
+            }
+
+            vdafObject = {
+                type: vdaf.type,
+                buckets: vdaf.buckets as string[],
+            }
+            measurement = body.measurement;
+            break;
+
+        default:
+            throw new Error(`Unrecognized VDAF ${vdaf.type}`);
+    }
+
+    return {
+        taskId: body.taskId,
+        leader: body.leader,
+        helper: body.helper,
+        vdaf: vdafObject,
+        measurement: measurement,
+        nonceTime: body.nonceTime,
+        minBatchDuration: body.minBatchDuration,
+    }
+}
+
+function assertUnreachable(_: never): never {
+    throw new Error("Unreachable code");
+}
+
+async function uploadHandler(req: Request, res: Response): Promise<void> {
+    try {
+        const body = sanitizeRequest(req.body);
+
+        // TODO (issue #97): Implement a way to construct a report with a
+        // custom nonce timestamp.
+        if (body.nonceTime !== undefined) {
+            throw new Error("`nonceTime` is not yet supported");
+        }
+
+        switch (body.vdaf.type) {
+            case "Prio3Aes128Count":
+                await new DAPClient({
+                    taskId: body.taskId,
+                    leader: body.leader,
+                    helper: body.helper,
+                    minBatchDurationSeconds: body.minBatchDuration,
+                    type: "count",
+                }).sendMeasurement(body.measurement != 0);
```

Comment:
Can this be `!==` or does this rely on string coercion intentionally in a way that I'm not seeing right now?

Reply:
Good catch, that was an oversight. The type is narrowed to `0 | 1` during input handling, (inside the other switch) but then it gets stored in the more permissive union.